### PR TITLE
[#578] Add binding for AcroFormDocumentHelper::setFormFieldName method

### PR DESCRIPTION
**Pull request by amargerandhw**
resolves https://github.com/pikepdf/pikepdf/issues/578

### CODE REVIEW

File: src/core/acroform.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "pikepdf.h"
+
+void init_acroform(py::module_ &m)
+{
+    py::class_<QPDFAcroFormDocumentHelper,
+        std::shared_ptr<QPDFAcroFormDocumentHelper>,
+        QPDFObjectHelper>(m, "AcroFormDocument")
```
Comment (jbarlow83) [inline src/core/acroform.cpp:22]: QPDFAcroFormDocumentHelper is a subclass of QPDFDocumentHelper, so the specification is simply

```
py::class_<QPDFAcroFormDocumentHelper,
        QPDFDocumentHelper>
```

File: src/pikepdf/_methods.py [hunk 2]
```diff
@@ -350,6 +351,7 @@ def open(
         access_mode: AccessMode = AccessMode.default,
         allow_overwriting_input: bool = False,
     ) -> Pdf:
+        print('open')
```
Comment (jbarlow83) [inline src/pikepdf/_methods.py:354]: Remove debug code

File: src/pikepdf/_methods.py [hunk 3]
```diff
@@ -690,6 +692,11 @@ def __hash__(self):
         return hash((self.llx, self.lly, self.urx, self.ury))
 
 
+@augments(AcroFormDocument)
+class Extend_AcroFormDocument:
+    def __repr__(self):
+        return f"<pikepdf._core.AcroFormDocument description>"
```
Comment (jbarlow83) [inline src/pikepdf/_methods.py]: This doesn't provide useful information. Which PDF, which form?

File: src/pikepdf/objects.py [hunk 4]
```diff
@@ -98,8 +98,8 @@ def __new__(cls, name: str | Name) -> Name:
         """Construct a PDF Name."""
         # QPDF_Name::unparse ensures that names are always saved in a UTF-8
         # compatible way, so we only need to guard the input.
-        if isinstance(name, bytes):
-            raise TypeError("Name should be str")
+        # if isinstance(name, bytes):
+        #     raise TypeError("Name should be str")
```
Comment (jbarlow83) [inline src/pikepdf/objects.py]: Remove debug code

File: src/core/acroform.cpp [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            py::init([](QPDF &q) {
+                QPDFAcroFormDocumentHelper afdh(q);
+
+                return afdh;
+            }),
+            py::keep_alive<0, 1>(), // LCOV_EXCL_LINE
+            py::arg("q")
+        )
+        .def(
+            "set_form_field_name",
```
Comment (jbarlow83) [inline src/core/acroform.cpp:33]: For this to be useful to anyone who say, wants to know what form fields are present before changing their names, you'll need to bind QPDFFormFieldObjectHelper as well. Then one should be able to retrieve form fields from the form and manipulate the field name from there.

I'm not going to accept a PR that solves your issue alone and isn't helpful to others. It needs to be of general use. All of the methods of QPDFFormFieldObjectHelper and QPDFAcroFormDocumentHelper should be exposed. You're probably going to need them anyway.

Please bind get/set functions as `def_property` which is equivalent to setting a `@property` with a pythonic getter and setter. Then one would use `form.fields[...].name = 'MyName'` etc.

**Review by jbarlow83 — COMMENTED** [review]
Please see other line item comments. Also, please add test cases to exercise any code you add.

(I didn't see this was a draft. Apologies if these remarks are premature.)

File: src/core/acroform.cpp (hunk 5 above)
Comment (amargerandhw) [reply src/core/acroform.cpp:26]: I added the method getFormFields, although it is mandatory for the usage that I solve. You could simplify use it as I showed in the test.
I dont know how I should do the binding with def_property.
If I understand you would me to add it to the binding of the class QPDFFormFieldObjectHelper but the method is on the class QPDFAcroFormDocumentHelper.

Implementing all the methods would take me too much time as I'm completely a novice in python and just discovered how pybind11 works, we could merge the base that I implemented and someone could improve it in the future

**Review by jbarlow83 — COMMENTED** [review]
Sorry, but I cannot accept this PR in its current form. 

As previously mentioned, get/set methods are not Pythonic and need to be replaced with pybind11 `def_property`. 

The implementation is also lacking many features others would need.

File: src/core/object.cpp [hunk 6]
```diff
@@ -895,6 +896,18 @@ void init_object(py::module_ &m)
         .def_property_readonly("obj", [](QPDFObjectHelper &poh) -> QPDFObjectHandle {
             return poh.getObjectHandle();
         });
+    py::class_<QPDFDocumentHelper, std::shared_ptr<QPDFDocumentHelper>>(m, "DocumentHelper");
+//        .def(
```
Comment (jbarlow83) [inline src/core/object.cpp:900]: Why is this commented out? It breaks existing functionality.

**Pull request closed without merging**